packages/core/src/module-resolver.ts

```diff
@@ -466,8 +466,15 @@ export class Resolver {
     } else {
       pkg = requestingPkg;
     }
-
-    return logTransition('entrypoint', request, request.virtualize(resolve(pkg.root, '-embroider-entrypoint.js')));
+    let matched = resolveExports(pkg.packageJSON, '-embroider-entrypoint.js', {
+      browser: true,
+      conditions: ['default', 'imports'],
+    });
+    return logTransition(
+      'entrypoint',
+      request,
+      request.virtualize(resolve(pkg.root, matched?.[0] ?? '-embroider-entrypoint.js'))
+    );
```

Comment:
This was a first attempt at fixing the way the entrypoint works, since it may need to crawl a different directory than the package root, given what's in package.json exports.

But this is not really sufficient by itself. The `AppFiles` class assumes that app and tests have already been smashed together.

Reply:
Putting the entrypoint inside a path that is mapped via package.json exports seems like a good idea, so this code is probably helpful. It ensures that we're resolving things that are actually supposed to be resolvable from outside the package. But we can't use only this path to identify which dirs to crawl when building the entrypoint, since there is more than one dir to crawl.

Instead, it probably makes sense to make the crawling code attempt to apply `resolve.exports` to `the-package/tests/example` and `the-package/example` to identify the two directories we need to crawl to build AppFiles.